syntax = "proto3";

package kv;

message SetValueRequest {
    string key = 1;
    string value = 2;
}

message SetValueResponse {
    enum Result {
        SUCC = 0;
        FAIL = 1;
    }
    Result result = 1;
}

message GetValueRequest {
    string key = 1;
}

message GetValueResponse {
    enum Result {
        Found = 0;
        NotFound = 1;
    }
    Result result = 1;
    string value = 2;
}

service KeyValue {
    rpc SetValue (SetValueRequest) returns (SetValueResponse);

    rpc GetValue (GetValueRequest) returns (GetValueResponse);
}
